source/server/core_server.cpp

```diff
@@ -272,10 +276,12 @@ int main(int argc, char** argv)
     schedParam.sched_priority = 95;
     sched_setscheduler(0, SCHED_FIFO, &schedParam);
 
-    cpu_set_t cpuSet;
-    CPU_ZERO(&cpuSet);
-    CPU_SET(6, &cpuSet);
-    sched_setaffinity(0, sizeof(cpu_set_t), &cpuSet);
+    if (config.cpu_affinity.server >= 0) {
```

Comment:
The `grpc-device` grpc server is multi-threaded, right? Should we be setting the affinity of grpc worker threads?

Reply:
OVERALL I don't think this feature is correct in how it handles the "server". If most users don't need/use the "server" affinity feature, maybe it is OK.

The reason someone would use "server" affinity, as I understand it, is to make sure that the server *DOES NOT* use the CPUS that are assigned to the moniker processing loop threads. The "server" should not be single-CPU, so it doesn't make sense to assign it a single CPU affinity.

I think what *WOULD* make sense: IF sideband and/or moniker-stream affinity are set, THEN set the server to use all CPUS EXCEPT the ones that are assigned to the processing threads. That could actually be the logic for `-1` for the processing threads as well (IF sideband affinity is set AND stream affinity is NOT SET, THEN stream is assigned to all other CPUs).

Reply:
@astarche I think what you are saying makes sense. If we set the cpu affinity of the server main thread to a specific thread, it just makes the server startup pinned on a specific thread. Subsequent server calls( whatever is served from 31763) from the client would anyways be  in specific threads. So this affinity would not do much except probably make server startup time better.

Reply:
Removed the configuration for server.